jabgui/src/main/java/org/jabref/gui/fieldeditors/MarkdownEditor.java

```diff
@@ -40,4 +64,90 @@ public void paste() {
+            Dragboard dragboard = event.getDragboard();
+            boolean success = false;
+            if (dragboard.hasFiles()) {
+                for (File file : dragboard.getFiles()) {
+                    if (FileUtil.isImage(file.toPath())) {
+                        success |= insertFileToLibraryAndAppend(file, textArea);
+                    }
+                }
+            } else if (dragboard.hasImage()) {
+                success |= insertImageToDbAndAppend(dragboard.getUrl(), textArea);
+            }
+            event.setDropCompleted(success);
+            event.consume();
+        });
+    }
+
+    /**
+     * Inserts Markdown Link for a file dropped into the editor, copying file to the database file directory if necessary
+     *
+     * @param file is the file that is dropped into the editor
+     * @param textArea is the Editor text area
+     * @return true if the Markdown Text was inserted, false otherwise
+     */
+    private boolean insertFileToLibraryAndAppend(File file, EditorTextArea textArea) {
+        Optional<Path> fileDir = databaseContext.getFirstExistingFileDir(filePreferences);
+        if (fileDir.isEmpty()) {
+            LOGGER.warn("No file directory found, cannot drop file");
+            return false;
+        }
+        List<Path> fileDirList = databaseContext.getFileDirectories(filePreferences);
+        Path relativePath = FileUtil.relativize(file.toPath(), fileDirList);
+
+        // If the path can be reached from the configured directories somehow, just insert the Markdown text
+        if (!relativePath.isAbsolute()) {
+            insertMarkdownText(relativePath.toString(), textArea);
+            return true;
+        }
+        Path destination = fileDir.get().resolve(file.getName());
+        String relativePathString = FileUtil.relativize(destination, fileDirList).toString();
+        try {
+            Files.copy(file.toPath(), destination);
+            insertMarkdownText(relativePathString, textArea);
+            return true;
+        } catch (FileAlreadyExistsException e) {
+            LOGGER.warn("Dropped file already exists: {} ", destination, e);
+            insertMarkdownText(relativePathString, textArea);
+            return true;
+        } catch (IOException e) {
+            LOGGER.error("Could not copy file: {} ", destination, e);
+            return false;
+        }
+    }
+
+    private boolean insertImageToDbAndAppend(String imageUrl, EditorTextArea textArea) {
+        try {
+            Path tempFile = new URLDownload(imageUrl).toTemporaryFile();
+            return insertFileToLibraryAndAppend(tempFile.toFile(), textArea);
+        } catch (MalformedURLException | FetcherException e) {
+            LOGGER.error("Failed to download image from URL: {}", imageUrl, e);
+            return false;
+        }
+    }
+
+    private void insertMarkdownText(String relativePath, EditorTextArea textArea) {
+        String markdownText = "![](" + relativePath + ")\n";
```

Comment:
Markdown only understands forward / slashes! Backslash \ is an escape character.  This might explain the path issue

https://www.markdownguide.org/basic-syntax/